src/NumericalAlgorithms/Spectral/ComplexDataView.hpp

```diff
@@ -0,0 +1,117 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <complex>
+#include <cstddef>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Spectral {
+namespace SWSH {
+
+/// \brief A storage container for storing references ("views") into parts of a
+/// ComplexDataVector.
+///
+/// \details Abstraction class which deals with the memory management for
+/// possible representations of complex numbers. Largely, this class behaves as
+/// a traditional "view" of the complex data, but can provide access to the data
+/// as separate real and imaginary component vectors, which are cached
+/// internally to save on copies where possible.
+class ComplexDataView {
```

Comment:
The consensus from @fmahebert and @markscheel was this will be easier to understand if it is a single class with some edits (largely removing the functions which return references to the internal vectors). The choice of the names for the enumeration which chooses between behavior is the main visible interface for calling code. The design for the enum should be:
```
enum class ComplexVectorRepresentation {
  RealsThenImags,
  Interleaved
}
```
The doxygen documentation should then be limited to the interface behavior of the functions, and not the internal details, such as when copies do or do not happen for the choices of memory representation. A summary of the general behavior for the two representations should be grouped with the enum values.  The full details are important for optimization, but should be placed in ordinary code comments in the cpp file instead.